### ISSUE TITLE
[UX] Views interface: "Show: Content" and "Show: Fields" may be confusing.

### ISSUE THREAD

**Issue by stpaultim** [backdrop/backdrop-issues#5985]
## Description of the bug

@keiserjb and I are at Florida Drupal Camp and saw a presenter shared this pet peeve in Drupal.
https://www.drupal.org/project/drupal/issues/3230442

We thought it would be fun to fix this in Backdrop before they do in Drupal, if there is agreement.

The issue is the word "Content" here when "Content Display Mode" would be much more appropriate or descriptive. 
![image](https://user-images.githubusercontent.com/3144571/219965991-d2726df8-938a-4278-89c3-de705bb217c4.png)

I think the problem is more apparent here:
![image](https://user-images.githubusercontent.com/3144571/219966092-1e2174f5-8b18-4675-a7ec-a402a7f59578.png)

This isn't a choice between Content or Fields. Fields are also content. It's a choice between using Content Display Modes to display the content or Fields. 

Not sure what other implications there are to this issue, but it seems like it might be an easy fix. Curious to see what others think.

**Comment by keiserjb** [comment]
![image](https://user-images.githubusercontent.com/41843284/219972722-0ce4c885-c4f6-473b-978d-45b081a8c5a8.png)

I don't know if this label change affects anywhere else within Backdrop.

**Comment by stpaultim** [comment]
This change makes sense to me. I'm a little uncertain of whether this change helps new users or not? I think it does, but welcome other opinions on this. From a developer perspective, I think it's clearer. 

Also not sure if there are implications elsewhere in the UI that I am not considering.

I've tested this and it works for me. I'm going to mark this as works for me, but would love for someone else to test as well.

**Comment by stpaultim** [comment]
We realized that Display Mode is the better term.

![image](https://user-images.githubusercontent.com/3144571/219975381-7247bf39-08f9-4b6b-bf51-7922222c6546.png)

**Comment by kiamlaluno** [comment]
To me, *Show: Content* is still preferable to *Show: Content Display Mode*, at least because *Content Display Mode* does not tell me what is shown.

What is called *Content* really is *Rendered content* or *Rendered nodes*, which means each node is rendered via `$node->view()`.

With *Show: Fields*, each of the listed fields is rendered singularly, probably not in the way `$node->view()` would render them.

**Comment by argiepiano** [comment]
> To me, Show: Content is still preferable to Show: Content Display Mode (which seems more a format type), at least because Content Display Mode does not tell me what is shown.

> What is called Content in that setting really is Rendered content 

I agree 100%. To me "Show: Content Display Mode" is really cryptic and bound to confuse someone who doesn't understand what a "Display Mode". "Rendered Content" on the other hand is pretty self explanatory.

**Comment by klonos** [comment]
Here's some feedback from me as well:
- _If_ we decide that this is a useful change, then we shouldn't be capitalizing things in camel case. It should be `Show: Content display mode` or `Show: Content view mode` - NOT `Show: Content Display Mode` or `Show: Content View Mode`.
- If we need to clarify that what is being used here is a display mode, then I propose something like this instead:
  ![image](https://user-images.githubusercontent.com/2423362/220147067-35f249e0-4bc2-4df3-b632-399c3d6a4740.png)
  ...or this:
  ![image](https://user-images.githubusercontent.com/2423362/220147220-b8249f67-532e-42ce-9e2a-e7a36de3c804.png)
- If we need to clarify things in the dialog, then we can use help text for the options. Something like this (rough text suggestion I whipped quickly, without much thought):
  <img src="https://user-images.githubusercontent.com/2423362/220151957-0b1cdde0-544d-40c8-9e4c-d57aca922be8.png" width=50%>

**Comment by stpaultim** [comment]
I appreciate the feedback. Here is a bit more context:

This pet peeve is coming from a very experience Drupal educator who spends his time teaching Drupal to people. While this is not an issue that has bothered me or anything I feel strongly about, the concern comes in that fields are also content. I suspect that this frustrates him, because it's confusing to explain to new Drupal users.

When you select fields, you are also seeing rendered content. You are seeing content rendered based upon the fields. The "label" fields tells you how you are seeing the content. The label "Content" tells you nothing about how your content is rendered and does suggest that "Content" is something different from "Fields." That doesn't make sense to me. 

Having said all of that, I'm personally open to altneratives such as those suggested by @klonos (they look good so far). I'm on my way back from Florida Drupal Camp and hoping to look closer at this when I get back.

**Comment by kiamlaluno** [comment]
I like @klonos suggestion. The dialog that appears when users click on *Content* or *Fields* should describe the difference between the choices.  
I still prefer *Rendered content* over *Content*, if that "label" is going to be changed.

As for choosing between the first or second screenshot, I prefer the first one.

**Comment by klonos** [comment]
I see your concerns @stpaultim and I don't have an ideal solution - only thoughts on how to improve things. It would be great if we got further feedback from the person that raised this concern, and what they think of our proposed improvements.

I believe that the key difference between these two options from a technical perspective is where the fields that will be used in the rendered view results are to be configured:
- the current "Content" option requires you to tweak the fields in a display mode on the content type
- the current "Fields" option allows you to tweak the fields within the view currently being edited

I have tried to convey that in the help text I used for the two options, but we all know that nobody reads help text 😓  ...especially if it's lengthy, and this is one of those cases where less words can be a cause for confusion/misunderstanding/inaccuracies.

> ...the concern comes in that fields are also content.

🤔 then perhaps another alternative could be this:
![image](https://user-images.githubusercontent.com/2423362/220544890-fc73960d-abb7-4cf6-8ed3-e7bbf42303f6.png)
![image](https://user-images.githubusercontent.com/2423362/220545139-8fab829c-63d2-47b2-b5c3-402f6d6e41cb.png)

**Comment by klonos** [comment]
@stpaultim I think that I will provide a PR, and then you could perhaps point that person to the PR sandbox, to test and tell us if things have improved.

**Comment by klonos** [comment]
Here's a PoC PR that people can play with: https://github.com/backdrop/backdrop/pull/4354

Code is not ready for review yet:
- There's inline comments with notes-to-self
- I've kept the original row style plugin texts alongside the new ones, but commented out.
- The code that generates the individual descriptions for each radio option is WIP - I've tried something that didn't work as well, and I'm thinking to revert it (in which case the code will be simplified)
- Tests will surely fail - lets fix those only if we decide that this is the best approach for now.

**Comment by BWPanda** [comment]
I'm not sure about this issue... I think the current wording makes sense, and the fact that (apparently) no one in Backdrop has brought this up as a problem makes me wonder what we're actually trying to solve...

In Backdrop, we substituted the word 'node' for 'content'. Consider this UI if you think of 'Content' instead as 'Node'. To me it makes sense - choose to display a node (and then refine by content type), or display fields (and then refine by the individual fields). So if you think of 'Content' in this way (as a node), then it works, and I don't see a problem with it.

Maybe this is confusing in Drupal because they did (do?) still use 'node' elsewhere, but 'content' here in Views. I don't know. I can speak to the Drupal issue. But it's not something I see a need to fix in Backdrop.

**Comment by kiamlaluno** [comment]
I do not find *Content* and *Fields* confusing, but I still think the dialog box that appears when choosing between one or the other should at least have a description similar to the following one (shown in a screenshot by @klonos).

![screenshot](https://user-images.githubusercontent.com/29851/222926979-f6e51eda-ef1a-409f-8333-e2111dc6c18f.png)

**Comment by argiepiano** [comment]
This may be just me, but the descriptions of that modal image seem very confusing. "Each [row] will be rendered as a display mode" seems very obscure to me.

**Comment by kiamlaluno** [comment]
@argiepiano Yes, the description can be better.
There should be a description, though, not just *Content* and *Fields*. I understand *Content* as *content as whole*, but I would not be able to give a description that is clear for everybody.

**Comment by klonos** [comment]
> ...the descriptions of that modal image seem very confusing.

@argiepiano in my alternative PR I have used the following for labels/descriptions:
![image](https://user-images.githubusercontent.com/2423362/223586187-34d465b3-a7a0-4005-8b6d-99674eecf300.png)

Do you find those to be an improvement, or still confusing? Any suggestions?

---
For the visually impaired, here's the text:
○ Content display mode
  > The fields rendered in each view result can only be managed in the selected content display mode.

○ Views-managed fields
  > The fields for each view result can be managed directly within this view.
